src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -767,8 +817,7 @@ int decryptFinal(ByteBuffer src, ByteBuffer dst)
         // scenario for the subsequent output buffer capacity check.
         checkDataLength(0, len);
 
-        if ((ibuffer.size() + ct.remaining()) - tagLenBytes >
-            dst.remaining()) {
+        if (len > dst.remaining()) {
             throw new ShortBufferException("Output buffer too small");
         }
```

Comment:
How is this different from the check at line 775-778 at the beginning of this method?

Reply:
Ah.. good.. I'll remove the one above from GCTR as I liked this check better and it protects the GCTR update ()